x/ccv/consumer/keeper/relay.go

```diff
@@ -25,8 +25,10 @@ func (k Keeper) OnRecvVSCPacket(ctx sdk.Context, packet channeltypes.Packet, new
 	// get the provider channel
 	providerChannel, found := k.GetProviderChannel(ctx)
 	if found && providerChannel != packet.DestinationChannel {
-		// VSC packet was sent on a channel different than the provider channel
-		return utils.OnRecvPacketOnUnknownChannel(ctx, k.scopedKeeper, k.channelKeeper, packet)
+		// VSC packet was sent on a channel different than the provider channel;
+		// this should never happen
+		panic(fmt.Errorf("VSCPacket received on unknown channel %s; expected: %s",
+			packet.DestinationChannel, providerChannel))
```

Comment:
I'm not sure I understand why this is a panic. From reading https://github.com/cosmos/interchain-security/issues/292 I understand that the consumer can receive packets from channels which arent OPEN on the provider. It sounds like recoverable behavior, not panic behavior.

Reply:
See https://github.com/cosmos/interchain-security/issues/292#issuecomment-1231334565

Reply:
I'm not sure if we should have this branch at all. Having it implies it can happen sometimes.

Reply:
Agree with Dan. Unless it can somehow happen, but it sounds like that would require an incorrect provider.